middlewares/validators/items.js

```diff
@@ -0,0 +1,34 @@
+const Joi = require("joi");
+
+const { TYPES } = require("../../constants/items");
+
+const validateItemsPayload = async (req, res, next) => {
+  const schema = Joi.object({
+    itemid: Joi.string().trim().required(),
+    itemType: Joi.string()
+      .uppercase()
+      .custom((value, helper) => {
+        if (!TYPES.includes(value)) return helper.message("Not a valid type");
+        return value;
+      }),
+
+    tagPayload: Joi.array().custom((values, helper) => {
+      for (const val of values) {
+        if (val.length !== 2) return helper.message("tag payload is not valid");
```

Comment:
not able to get what we are validating here, can you please explain

Reply:
Here we are checking if the `tagpayload` has array elements of `length 2` comprising `tagid` and `levelid`. In case one of them is missing, `tag[1]` will fail in the model. 
For tags without levels, `tag[1]` will be ""

Reply:
Here you are checking length, but how will you figure out there is tagid and levelid because here you are checking the length of node or keys?
You should check whether both nodes exist or not.

Reply:
Makes sense. Fixed the payload schema